Dockerfile

```diff
@@ -53,16 +53,19 @@ RUN echo "@cloudposse https://apk.cloudposse.com/3.9/vendor" >> /etc/apk/reposit
 # Use TLS for alpine default repos
 RUN sed -i 's|http://dl-cdn.alpinelinux.org|https://alpine.global.ssl.fastly.net|g' /etc/apk/repositories && \
     echo "@testing https://alpine.global.ssl.fastly.net/alpine/edge/testing" >> /etc/apk/repositories && \
-    echo "@community https://alpine.global.ssl.fastly.net/alpine/edge/community" >> /etc/apk/repositories && \
-    apk update
+    echo "@community https://alpine.global.ssl.fastly.net/alpine/edge/community" >> /etc/apk/repositories
 
 # Install alpine package manifest
 COPY packages.txt /etc/apk/
+# Install repo checksum in an attempt to ensure updates bust the Docker build cache
+COPY package_repos.md5 /var/cache/apk/
```

Comment:
Have you tried doing this instead:

```
ADD https://apk.cloudposse.com/3.9/vendor/x86_64/APKINDEX.tar.gz /var/tmp/APKINDEX.tar.gz
```
(or the like?)

My underestanding is `ADD` will evaluate the contents of the endpoint to determine if the layer has changed.

Reply:
Also, while `COPY` is generally the best practice, it doesn't mean using `ADD` is wrong; there's a time and place for it.

Reply:
You have to do that for each package repo, and it executes on every build, which slows me down a lot (okay, not a lot, but it's annoying). It also yields the opposite problem to the one we have now. Right now, we are missing package updates we want because build layers are cached. With ADD, we will get package updates as soon as they are released whether we want them or not, and without warning or notification.

Reply:
> With ADD, we will get package updates as soon as they are released whether we want them or not, and without warning or notification.

Agree. 

> With the explicit make of the md5 file, we get something closer to npm, bundler, bazel where the packages only change when we explicitly ask them to. This method is far from foolproof, but is at least a good compromise in that direction.

Agree. It's a step in the right direction.